javaslang-benchmark/src/test/java/javaslang/idiom/ForBenchmark.java

```diff
@@ -0,0 +1,66 @@
+package javaslang.idiom;
+
+import javaslang.JmhRunner;
+import javaslang.collection.Array;
+import javaslang.collection.Iterator;
+import org.junit.Test;
+import org.openjdk.jmh.annotations.*;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static javaslang.API.For;
+import static javaslang.JmhRunner.getRandomValues;
+
+public class ForBenchmark {
+    static final Array<Class<?>> CLASSES = Array.of(
+            ForComprehension.class
+    );
+
+    @Test
+    public void testAsserts() {
+        JmhRunner.runDebugWithAsserts(CLASSES);
+    }
+
+    public static void main(String... args) {
+        JmhRunner.runNormalNoAsserts(CLASSES);
+    }
+
+    @State(Scope.Benchmark)
+    public static class ForComprehension {
+        @Param({ "10"/*, "100", "1000" */ })
+        public int CONTAINER_SIZE;
+
+        int EXPECTED_AGGREGATE;
+        Integer[] ELEMENTS;
+        List<Integer> ELEMENTS_ARRAY;
+
+        @Setup
+        public void setup() {
+            ELEMENTS = getRandomValues(CONTAINER_SIZE, 0);
+            ELEMENTS_ARRAY = Arrays.asList(ELEMENTS);
+            EXPECTED_AGGREGATE = Iterator.of(ELEMENTS).reduce(JmhRunner::aggregate);
+        }
+
+        @Benchmark
+        public Object javaslang_for() {
+            Integer result = For(ELEMENTS_ARRAY).yield(i -> i).reduce((i, j) -> i + j);
+            assert result.intValue() == EXPECTED_AGGREGATE;
+            return result;
+        }
+
+        @Benchmark
+        public Object java_for() {
+            Integer result = 0;
+            ELEMENTS_ARRAY
```

Comment:
you're absolutely right, like I said just wanted to get a feel of it. In particular I never used the `for` comprehension so I would not know what would be the right way. Do you have any suggestions on how to properly test it? My rationale was that reduce code was the same in both cases (broken logic I know).

The for comprehension acts both as a map and as a consumer, so I dont exactly know what's the best way to test.